nexus/db-queries/src/db/datastore/region_snapshot.rs

```diff
@@ -120,4 +120,40 @@ impl DataStore {
             .await
             .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
     }
+
+    /// Find region snapshots not on expunged disks that match a snapshot id
+    pub async fn find_non_expunged_region_snapshots(
```

Comment:
Yeah, the inverse of this would be finding all expunged region snapshots, which something like the region snapshot replacement start background task does do. But I'm not sure about including both